src/merkle/mod.rs

```diff
@@ -27,6 +27,12 @@ pub use path::{MerklePath, RootPath, ValuePath};
 mod simple_smt;
 pub use simple_smt::SimpleSmt;
 
+mod smt;
+pub use smt::{InnerNode, LeafIndex, SparseMerkleTree, SMT_MAX_DEPTH, SMT_MIN_DEPTH};
+
+mod new_smt;
+pub use new_smt::{NewSmt, NewSmtKey, NewSmtLeaf, NEW_SMT_DEPTH};
```

Comment:
If we keep `SparseMerkleTree` private in this create, we probably won't need to make `InnerNode` and `LeafIndex` public either.

Reply:
Made `SparseMerkleTree` private, but let `LeafIndex` public, since `LeafIndex`: is now part of `SimpleSmt`'s public API (e.g. [`get_leaf()`](https://github.com/0xPolygonMiden/crypto/blob/a6068001828af3c2e2e041b819fb930d1467aac2/src/merkle/simple_smt/mod.rs#L117))